### DISCUSSION TITLE
How to shuffle a page collection?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#285] [Ask a question] [answered]
Hello, 

I found an example snippet in the API documentation to show random pages and would like to use it in a plugin as a shortcut to have more flexibility. This is what I tried so far: 

```php
		if($name=="random" && $shortcut)
		{
			list($location, $pagesMax) = $this->yellow->toolbox->getTextArgs($text);
			if(empty($location)) $location = $this->yellow->config->get("randomLocation");
			if(strempty($pagesMax)) $pagesMax = $this->yellow->config->get("randomPagesMax");
			$pages = $this->yellow->pages->find($location);
			$output .= "<div class=\"".$name."\">\n";
			foreach($this->yellow->pages->shuffle()->limit($pagesMax) as $page) {
			// Page list
```

I'll leave the rest of the code aside for the moment. The above lines result in the following error: 

```
Fatal error: Call to undefined method YellowPages::shuffle() in /[...]/system/plugins/random.php on line 31 
``` 

Any idea what's going wrong in my guesswork code and how to use the page shuffle function correctly? What I'm trying to do in result is a random page selection from a specific location, assuming this is possible actually. 

Thanks in advance. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
Hello Steffen,

Here's an example to access random pages at a specific location:

```php
$pageFound = $this->yellow->content->find($location);
$pages = $pageFound ? $pageFound->getChildren() : $this->yellow->content->clean();
foreach($pages->shuffle()->limit(3) as $page) {
  // More code here
}
```

A few more words how this works. First line gets the page for a location. Second line checks if this page exists and then gets all child pages. Third line randomises these pages and limits the number of them. As you can see in the [API](https://datenstrom.se/yellow/help/api-for-developers) the function `shuffle()` works with a Yellow page collection.

Tested with Datenstrom Yellow 0.8.22. Hope this helps. 😄

**Comment by schulle4u** [comment]
Thanks Mark, exactly what I was looking for. I'll need to tweak some things, but the plugin already works. :)

Steffen
